test-integration/reference-book.coffee

```diff
@@ -9,68 +10,111 @@ SECTIONS_TO_TEST = 10
 
 describe 'Reference Book Exercises', ->
 
-  @it 'Loads Biology reference book (readonly)', ->
-    @login(TEACHER_USERNAME)
+  @it 'Loads Biology and Physics reference book, and checks for missing exercises (readonly)', ->
 
+    allMissingExercises = []
 
-    checkForMissingExercises = =>
+    checkSectionsForMissingExercises = =>
       # Wait until the book has loaded.
-      @addTimeout(60)
       @waitAnd(css: '.page-wrapper .page.has-html')
+        .then(closeMenuAndResizeBook)
+        .then =>
+          # Perform `checkEachPage` in series for SECTIONS_TO_TEST times.
+          @forNTimesInSeries(SECTIONS_TO_TEST, checkEachPage)()
+
+    closeMenuAndResizeBook = =>
+      # Resize prevents need to click next again -- without this,
+      # if the window is too skinny/if the next button is right by the scroll bar,
+      # the page will accidentally get clicked to scroll as opposed to clicking next.
+      @driver.manage().window().setSize(1080, 1080)
```

Comment:
Could this be a helper? And instead of setting the size explicitly, have methods like `.setIPhone()`, `.setTablet()`, `.setNarrow()`, `.setLarge()` (not suggesting all would be implemented). That way we would have some "standard" sizes we'd be testing with?